src/cli/ionic-config.ts

```diff
@@ -0,0 +1,45 @@
+import { readJSON, writeJSON, mkdirp } from '@ionic/utils-fs';
+
+import * as os from 'os';
+import * as path from 'path';
+
+import { uuidv4 } from './telemetry/helpers';
+
+export const CONFIG_FILE = 'config.json';
+export const DEFAULT_CONFIG_DIRECTORY = path.resolve(os.homedir(), '.ionic', CONFIG_FILE);
+
+export interface SystemConfig {
+	"telemetry"?: boolean,
+	"telemetry.stencil"?: boolean,
+	"tokens.telemetry"?: string,
+}
+
+export async function readConfig(): Promise<SystemConfig> {
+	try {
+		return await readJSON(DEFAULT_CONFIG_DIRECTORY);
+	} catch (e) {
+		if (e.code !== 'ENOENT') {
+			throw e;
+		}
+
+		const config: SystemConfig = {
+			"tokens.telemetry": uuidv4(),
+			"telemetry": true,
+			"telemetry.stencil": true,
+		};
+
+		await writeConfig(config);
```

Comment:
Are there any error cases from writing a config file we need to watch out for here? e.g. if we don't have permission to write to the local FS? I'm not familiar with this module, apologies if this is a dumb question 😆

Reply:
I don't doubt that there are edge cases, I'm just as unfamiliar with the package as you :) I'll add some try catch blocks, but also this shouldn't error during any normal Stencil usage, only unless the customer specifically fires a telemetry command.

Reply:
Added error handling inside this function definition